third_party/suitesparse.BUILD

```diff
@@ -0,0 +1,600 @@
+    ],
+    hdrs = [
+        "AMD/Include/amd.h",
+        "AMD/Include/amd_internal.h",
+    ],
+    includes = [
+        "AMD/Include",
+    ],
+    strip_include_prefix = "AMD/Include",
+    deps = [ "suitesparseconfig", ],
+    linkopts = [
+        "-lm",
+        "-lpthread",
+        "-lblas",
+        "-llapack",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "camd",
+    srcs = [
+        "CAMD/Source/camd_l1.c",
+        "CAMD/Source/camd_l2.c",
+        "CAMD/Source/camd_l_aat.c",
+        "CAMD/Source/camd_l_control.c",
+        "CAMD/Source/camd_l_defaults.c",
+        "CAMD/Source/camd_l_dump.c",
+        "CAMD/Source/camd_l_info.c",
+        "CAMD/Source/camd_l_order.c",
+        "CAMD/Source/camd_l_postorder.c",
+        "CAMD/Source/camd_l_preprocess.c",
+        "CAMD/Source/camd_l_valid.c",
+    ],
+    textual_hdrs = [
+        "CAMD/Source/camd_1.c",
+        "CAMD/Source/camd_2.c",
+        "CAMD/Source/camd_aat.c",
+        "CAMD/Source/camd_control.c",
+        "CAMD/Source/camd_defaults.c",
+        "CAMD/Source/camd_dump.c",
+        "CAMD/Source/camd_info.c",
+        "CAMD/Source/camd_order.c",
+        "CAMD/Source/camd_postorder.c",
+        "CAMD/Source/camd_preprocess.c",
+        "CAMD/Source/camd_valid.c",
+    ],
+    copts = [
+        "-Iexternal/suitesparse/CAMD/Source",
+    ],
+    hdrs = [
+        "CAMD/Include/camd.h",
+        "CAMD/Include/camd_internal.h",
+    ],
+    includes = [
+        "CAMD/Include",
+    ],
+    strip_include_prefix = "CAMD/Include",
+    deps = [ "suitesparseconfig", ],
+    linkopts = [
+        "-lm",
+        "-lpthread",
+        "-lblas",
+        "-llapack",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "colamd",
+    srcs = [
+        "COLAMD/Source/colamd_l.c",
+    ],
+    textual_hdrs = [
+        "COLAMD/Source/colamd.c",
+    ],
+    copts = [
+        "-Iexternal/suitesparse/COLAMD/Source",
+    ],
+    hdrs = [
+        "COLAMD/Include/colamd.h",
+    ],
+    includes = [
+        "COLAMD/Include",
+    ],
+    strip_include_prefix = "COLAMD/Include",
+    deps = [ "suitesparseconfig", ],
+    linkopts = [
+        "-lm",
+        "-lpthread",
+        "-lblas",
+        "-llapack",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "ccolamd",
+    srcs = [
+        "CCOLAMD/Source/ccolamd_l.c",
+    ],
+    textual_hdrs = [
+        "CCOLAMD/Source/ccolamd.c",
+    ],
+    copts = [
+        "-Iexternal/suitesparse/CCOLAMD/Source",
+    ],
+    hdrs = [
+        "CCOLAMD/Include/ccolamd.h",
+    ],
+    includes = [
+        "CCOLAMD/Include",
+    ],
+    strip_include_prefix = "CCOLAMD/Include",
+    deps = [ "suitesparseconfig", ],
+    linkopts = [
+        "-lm",
+        "-lpthread",
+        "-lblas",
+        "-llapack",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "GKlib",
+    defines = [
+        "LINUX",
```

Comment:
Does this imply this dependency only works on linux?

Reply:
Good question.  This value comes from https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/dev/CHOLMOD/SuiteSparse_metis/GKlib/GKlibSystem.cmake#L24 which defines `LINUX` as long as neither `MSVC` nor `MINGW` matches.  In spite of this, as far as I know, there's no problem running CHOLMOD/METIS on Macintosh.